crates/matrix-sdk-base/src/store/traits.rs

```diff
@@ -390,6 +391,10 @@ pub trait StateStore: AsyncTraitDeps {
     ) -> Result<bool, Self::Error>;
 
     /// Loads all the send queue requests for the given room.
+    ///
+    /// The resulting vector of queued requests should be ordered from higher
+    /// priority to lower priority, and respect the insertion order when
+    /// priorities are equal.
     async fn load_send_queue_requests(
```

Comment:
Ah maybe I understand: indeed, although we load the entire list (we expect it to be mostly empty, most of the time), we do *not* remove it from the storage when we do so, and only keep it transiently in memory. So there's no risk of missing something to be sent, as far as I can tell. If you were raising another problem, please let me know :pray:

Reply:
Clarified out of band:

I was worried that we would:

* Fetch all queued requests from the store
* Process all those requests, during which we add the new high-priority request to the store
* Fetch requests from the store but it would be too late because we'd already processed everything

but in fact, we:

* Fetch all queued requests from the store
* Ignore most of them, and just process the first one, during which we add the new high-priority one
* Repeat, meaning that the next one that is processed is the new high-priority one, as we want.

So it's all fine.

And @bnjbvr confirmed that it is indeed all fine, and he plans some refactoring that will help make this easier to understand.